### DISCUSSION TITLE
如何实现路由规则的匹配？

### DISCUSSION THREAD

**Discussion by Cyberbolt** [xjasonlyu/tun2socks#322] [Q&A]
首先，非常感谢作者对该项目的贡献！

请问如何实现路由规则的匹配？如中国大陆内 ip 走直连，大陆外走 socks。只有自己设置路由表来实现吗？如果 ip 数量有数以百万计，会对 tun2socks 的性能产生影响吗？

**Comment by nange** [comment]
我目前是在socks服务内部实现这个逻辑的。 自己去实现，灵活一点。

**Reply by Cyberbolt** [reply]
socks 在本地，实现不了，会成流量死循环，所有走 socks 又直连的流量又会回到 tun2socks

**Reply by nange** [reply]
如果你是把tun2socks和socks程序分开的作为两个程序的话，确实不行。不过如果把tun2socks作为库，自己开发的话，就是能实现的。可以参考我写的：https://github.com/nange/easyss  其中的获取直连连接：https://github.com/nange/easyss/blob/7da6fd6bb7a47bb4bc79e1bb5cfe036aa30609d8/direct.go#L71C4-L71C4

**Reply by Cyberbolt** [reply]
谢谢！不会 Golang，有没有只实现了选择性代理功能的 tun2socks？或者，直接在路由表上配置数以万计的 ip 会影响效率吗

**Reply by nange** [reply]
不太清楚，不过easyss就实现了你说的功能😃
